lib/product-import.coffee

```diff
@@ -230,6 +232,52 @@ class ProductImport
       else
         false
 
+  _updateProductRepeater: (prodToProcess, existingProduct) ->
+    repeater = new Repeater {attempts: 5}
+    repeater.execute =>
+      @_updateProduct(prodToProcess, existingProduct)
+    , (e) =>
+      if e.statusCode isnt 409 # concurrent modification
+        return Promise.reject e
+
+      @logger.warn "Recovering from 409 concurrentModification error on product '#{existingProduct.id}'"
+
+      Promise.resolve => # next task must be a function
+        @client.productProjections.staged(true).byId(existingProduct.id).fetch()
+        .then (result) =>
+          @_updateProduct(prodToProcess, result.body)
+
+  _updateProduct: (_prodToProcess, existingProduct) ->
+    prodToProcess = _.deepClone(_prodToProcess)
```

Comment:
Because the `prodToProcess` object is modified by `_prepareUpdateProduct` function which resolves several references and adds some metadata.

Maybe we can add a flag which will tell us that the object is already prepared for an update so we do not have to resolve references again and then we can remove that deepClone.